File: 2-algorithm/intermediateJs/convertHTMLents.js
/*jslint node: true */
'use strict';

function convertHTML (str) {
  // &colon;&rpar
  str = str.split('');
  for (var i = 0; i < str.length; i++) {
    str[i] = convert(str[i]);
  }
  return str.join('');
}

function convert (str) {
  if (str === '<') {return '&lt;';}
  if (str === '>') {return '&gt;';}
  if (str === '&') {return '&amp;';}
  if (str === '"') {return '&quot;';}
  if (str === "'") {return '&apos;';}
  return str;
}

console.log(convertHTML('Dolce & Gabbana'));

File: 2-algorithm/intermediateGo/convertHTMLents.go
package main

import (
	"fmt"
	"strings"
)

func convertHTML(str string) string {
	var res string
	letters := strings.Split(str, "")
	for _, v := range letters {
		v = convert2(v)
		res += v
	}
	return res
}

func convert2(str string) string {
	if str == "<" {
		return "&lt;"
	}
	if str == ">" {
		return "&gt;"
	}
	if str == "&" {
		return "&amp;"
	}
	if str == "\"" {
		return "&quot;"
	}
	if str == "'" {
		return "&apos;"
	}
	return str
}

func convertHTMLents() {
	fmt.Println(convertHTML("Dolce & Gabbana"))
	fmt.Println(convertHTML("Hamburgers < Pizza < Tacos"))
	fmt.Println(convertHTML("Sixty > twelve"))
	fmt.Println(convertHTML("Stuff in \"quotation marks\""))
	fmt.Println(convertHTML("Shindler's List"))
	fmt.Println(convertHTML("<>"))
	fmt.Println(convertHTML("abc"))
}
